modules/tests/test-integration/src/test/java/org/ballerinalang/test/context/ServerInstance.java

```diff
@@ -441,33 +443,28 @@ private String getServerPID() throws BallerinaTestException {
             }
             tmp.destroy();
         } else {
-            BufferedReader bufferedReader = null;
-            FileReader fileReader = null;
-
+            //reading the process id from netstat
+            Process tmp;
             try {
-                //reading the pid form  carbon.pid file in server home dir
-                fileReader = new FileReader(serverHome + File.separator + Constant.SERVER_PID_FILE_NAME);
-                bufferedReader = new BufferedReader(fileReader);
-                pid = bufferedReader.readLine();
+                tmp = Runtime.getRuntime().exec("lsof -Pi tcp:" + httpServerPort);
```

Comment:
Added that thanks @yasassri  for the point on how to use grep within exec command